bgpd/bgpd.c

```diff
@@ -4023,6 +4023,64 @@ static void peer_flag_modify_action(struct peer *peer, uint32_t flag)
 		bgp_session_reset(peer);
 }
 
+/* Enable global administrative shutdown of all peers of BGP instance */
+void bgp_shutdown_enable(struct bgp *bgp, char *msg)
+{
+	struct peer *peer;
+	struct listnode *node;
+
+	/* do nothing if already shut down */
+	if (CHECK_FLAG(bgp->flags, BGP_FLAG_SHUTDOWN))
+		return;
+
+	/* informational log message */
+	zlog_info("Enabled administrative shutdown on BGP instance AS %u",
```

Comment:
IMO, this should be under zlog_debug()

Reply:
I actually discussed this with @rzalamena and we concluded, that, since this would be a conscious action by a user, it should be regarded as informational and logged as such in accordance with the documentation of the [logging facilities](http://docs.frrouting.org/projects/dev-guide/en/latest/logging.html#informational-messages).